stable/datadog/templates/agent-deployment.yaml

```diff
@@ -0,0 +1,103 @@
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+spec:
+  replicas: {{ .Values.clusterAgent.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "datadog.clusterAgent.fullname" . }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "datadog.clusterAgent.fullname" . }}
+      name: {{ template "datadog.clusterAgent.fullname" . }}
+    spec:
+      {{- if .Values.clusterAgent.image.pullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.clusterAgent.image.pullSecrets | indent 8 }}
+      {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.clusterAgent.image.repository }}:{{ .Values.clusterAgent.image.tag }}"
+        imagePullPolicy: {{ .Values.clusterAgent.image.pullPolicy }}
+        resources:
+{{ toYaml .Values.clusterAgent.resources | indent 12 }}
+        ports:
+        - containerPort: 5005
+          name: agentport
+          protocol: TCP
+        {{- if .Values.clusterAgent.metricsProvider.enabled }}
+        - containerPort: 443
+          name: metricsapi
+          protocol: TCP
+        {{- end }}
+        env:
+          - name: DD_API_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "datadog.apiSecretName" . }}
+                key: api-key
+          {{- if .Values.clusterAgent.metricsProvider.enabled }}
+          - name: DD_EXTERNAL_METRICS_PROVIDER_ENABLED
+            value: {{ .Values.clusterAgent.metricsProvider.enabled | quote }}
+          - name: DD_APP_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "datadog.appKeySecretName" . }}
+                key: app-key
+          {{- end }}
+          {{- if .Values.datadog.logLevel }}
+          - name: DD_LOG_LEVEL
+            value: {{ .Values.datadog.logLevel | quote }}
+          {{- end }}
+          {{- if .Values.datadog.leaderElection }}
+          - name: DD_LEADER_ELECTION
+            value: {{ .Values.datadog.leaderElection | quote}}
+          {{- end }}
+          {{- if .Values.datadog.leaderLeaseDuration }}
+          - name: DD_LEADER_LEASE_DURATION
+            value: {{ .Values.datadog.leaderLeaseDuration | quote }}
+          {{- end }}
+          {{- if .Values.datadog.collectEvents }}
+          - name: DD_COLLECT_KUBERNETES_EVENTS
+            value: {{ .Values.datadog.collectEvents | quote}}
+          {{- end }}
+          - name: DD_CLUSTER_AGENT_AUTH_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "datadog.clusterAgent.fullname" . }}
+                key: token
+          - name: DD_KUBE_RESOURCES_NAMESPACE
+            value: {{ .Release.Namespace }}
+{{- if .Values.clusterAgent.env }}
+{{ toYaml .Values.clusterAgent.env | indent 10 }}
+{{- end }}
+        {{- if .Values.clusterAgent.metricsProvider.enabled }}
+        livenessProbe:
```

Comment:
probing the metrics provider is a great idea, we are currently rethinking how we do liveness probes, I will definitely keep this in mind.